src/Concerns/AppendsAttributesToResults.php

```diff
@@ -23,11 +23,51 @@ public function allowedAppends($appends): self
     }
 
     protected function addAppendsToResults(Collection $results)
-    {
-        return $results->each(function (Model $result) {
-            return $result->append($this->request->appends()->toArray());
-        });
-    }
+	{
+	    $appends = $this->request->appends();
+	    return $results->each(function($item) use($appends)
+	    {
+	        $appends->each(function($append) use($item)
+	        {
+	            if(strpos($append, '.'))
+	            {
+		        	$subs = collect(explode('.', $append));
+		        	$relation = $subs->shift();
+
+		        	$item = $this->appendLoop($item, $relation, $subs);
+	            } else
+	            {
+	                $item->append($append);
+	            }
+	        });
+	    });
+	}
+
+	private function appendLoop($item, $relation, $subs)
+	{
+		if($item->$relation !== null)
```

Comment:
I think this will cause the relationship to be loaded, even though it wasn't requested (or allowed). Can we use `$item->relationLoaded($relation)` instead?